frontend/src/store/teachers.js

```diff
@@ -87,4 +87,42 @@ export const teacherGetters = {
   teachers(state) {
     return state.teachers;
   },
+
+  suggestedTeachers(state) {
+    return (studentId) => {
+
+      // 1. Find the student using the ID
+      const student = state.students.find(student => student.StudentID === studentId)
+      if (!student) {
+        return []
+      }
+      // 2. Find their native language
+      const studentNativeLanguage = student.NativeLanguageID
+      
+      // 3. Get unmatched teachers whose first or second language equals student's native language
+      const relevantTeachers = state.teachers.filter(teacher => {
+        const doesNativeLanguageMatch = teacher.NativeLanguageID === studentNativeLanguage
+        const doesSecondLanguageMatch = teacher.SecondLanguageID === studentNativeLanguage
+        const isUnmatched = teacher.status.Description === "UNMATCHED"
```

Comment:
one teacher can be matched to multiple students, so we have to include teachers who are already `MATCHED` here.

Reply:
We made changes to include Matched teachers, but we also added a bit more sorting logic to favour unmatched teachers. Otherwise, as we match up earlier teachers with students, they will end up **always** being in the suggested teachers list (simply because they joined earlier).